### ISSUE TITLE
BattleEye Fails startup on Linux using latest Proton

### ISSUE THREAD

**Issue by DGxInfinitY** [ValveSoftware/Proton#30]
BattleEye for PlayerUnknowns BattleGrounds won't start on Elementary OS(Ubuntu Based)
As far as I know, this is a wine definitions problem that might require a little deep searching to fix.

**Comment by smurfhunter** [comment]
This should also affect Planetside 2, and I'd really love to see it working

**Comment by ghost** [comment]
BattlEye currently doesn't function at all on wine, however there is [someone on reddit](https://www.reddit.com/r/wine_gaming/comments/98yxv8/battleye_update_i_just_got_fortnite_running_on/) working on fixing that.

**Comment by linkjay** [comment]
I understand this isn't Proton's problem, but I think this is actually a crucial thing that should be fixed with Proton. This has been an issue with Wine for a long time and it seems like no one has really developed a fix for battleye. Considering how many games run the anticheat, I think it would be necessary for the project to have this fixed one way or another.

**Comment by adds68** [comment]
Also seeing this issue!

**Comment by jarrard** [comment]
I believe this might also be a issue for ARMA3, it does have BattleEye for Linux with a test build of the game but they lack a few features such as launcher options and MP compat with Windows users.

**Comment by ghost** [comment]
It was stated in the steam announcement post that some anticheats would render impossible the compatibility. BattleEye is so aggressive that they even blocked reshade a few months back. Doubt we will ever see those titles work with wine.

**Comment by mkljczk** [comment]
Maybe some guys (pun unintended) from Valve or CodeWeavers could help Guy1524 in making BattlEye work on wine, maybe they would work together?

**Comment by radude** [comment]
Same for DayZ standalone.

**Comment by oleksandrzelentsov** [comment]
This is the same for the PUBG.
I think Battleye should open up a little bit since it's a big problem for a lot of games, which makes it impossible to play them.

**Comment by Guy1524** [comment]
Hello, I would like to mention a few things:

Right now I am focused on submitting my current Battleye work to wine before I proceed.  Also, now that school is started progress will generally be slower, as I am only working on the weekends.

@oleksandrzelentsov People have tried contacting Battleye about this issue before, and their position is that they will only support native linux games.

@m4sk1n Plagman (valve dev) has expressed interest in working with anticheat developers to get it working under proton.  This solution would probably be more sustainable in the long term.

**Comment by Lyle-Tafoya** [comment]
Given that Battleye has native support on Linux, maybe the wizards over at Valve can find a way to do some kind of Battleye passthrough for Steam Play games to connect it with the native version of Battleye.  Running Battleye in Wine would be nice, but I am afraid that could also lead to people being banned without cheating.

**Comment by linkjay** [comment]
@Lyle-Tafoya Hmm, that's actually pretty interesting. I think it would also be worthy to reach out to the Battleye team and see if they will make room for Wine in their support so we can officially have support for the anticheat on their side. 

This way, we can avoid false-positives, and have good support in Wine. Now, we just have to convince them to actually go through with that...

**Comment by h8672** [comment]
Planetside 2 gets to the launcher now, battleye though... tries for an undetermined long time to launch it and the log file has various dlls showing there. I added some through winetricks and those dlls dont really show at the end of log file anymore. Can't find all dlls that show there in winecfg.
env WINEPREFIX="/mnt/tera/Games/SteamLibrary/steamapps/compatdata/218230/pfx" WINEPATH="/mnt/tera/Games/SteamLibrary/steamapps/common/Proton 3.7/dist/bin" winetricks

**Comment by dreamer** [comment]
I understand the sentiment in this issue (I would love to play Planetside 2 again), but trying to work around anti-cheat system is pointless - they will be updated faster than workarounds for Linux will appear. The only way forward would be to start a conversation with [Battleye devs](https://www.battleye.com/) about supporting Proton officially. They already can detect, that game is running inside Wine.

**Comment by addeps3** [comment]
I assume everyone understands that anti cheating software has so high expectations on it, and requires so much trust that it excludes practically everyone, except the industry to maintain it. 
So until the industry takes the initiative to create a true, cross platform anti-cheat, then I expect nothing to happen. :)

**Comment by HonkingGoose** [comment]
Implementation of some kind of whitelist/integration of anti-cheat systems into Proton is tracked here: #1468

**Comment by Guy1524** [comment]
I'd like to provide an update on the state of my battleye work.  A month ago, I got the battleye driver's anticheat loop running.  Once the game client attempts to connect with the driver and submits an interupt, there is an internal error, and the game I am testing refuses to proceed.  I am currently taking a break, but if anyone else experienced wants to pick up where I left off, my work is here: https://github.com/Guy1524/wine/tree/battleye-work

**Comment by ghost** [comment]
No news on it ? would like to see player unknow battelground but battleye is a pain in the ass

**Comment by Guy1524** [comment]
@edneyhelene FWIW even when battleye starts to work, PUBG has another preventing it from launching.

**Comment by elevenpassin** [comment]
I am really missing out playing Planetside 2. The battleeye anticheat starts but keeps on loading and the game never launches :(

**Comment by JanGalek** [comment]
Now cant play ATLAS on official servers.

Tested on Proton 3.7-8, 3.7-8 Beta, 3.16-4, 3.16-6 Beta, 4.2-9, 4.11-2.

**Comment by kisak-valve** [comment]
Merge with battleye patches that are being worked on in a separate repo

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2278.
@redpiller posted on 2019-01-28T17:36:04:

# Feature Request
https://github.com/Guy1524/wine/commits/battleye-work-stable
this repo contains wine patches that enables the use of battleye anticheat in games like
Arma2, Arma3, Fortnite, Planetside 2 ... and so on...
take a look, maybe you could establish a cross repo patching pipeline 

## I confirm:
- [ x ] that I haven't found another request for this feature.
- [ x ] that I have checked whether there are updates for my system available that
      contain this feature already.

## Description 
Battleye support

## Justification [optional] 
Battleye is broken in proton

## Risks [optional]
Not every game will be supported at first

## References [optional] 
https://github.com/Guy1524/wine/commits/battleye-work-stable

**Comment by mkljczk** [comment]
@redpiller, these patches don’t make BattlEye work, they just make it fail in further steps…

**Comment by HonkingGoose** [comment]
It seems BattlEye support for Linux will not be coming anytime soon:

https://www.gamingonlinux.com/articles/seems-like-theres-no-hope-for-battleye-support-within-steam-play.13747

Quote from BattlEye from gamingonlinux:
>Unfortunately we can only support Linux if the respective game natively does so.

**Comment by philipjohnbarlow** [comment]
Person here claims to have gotten PUBG to work by running it via the windows version of steam.

https://www.youtube.com/watch?v=Aa_TwN-daEA

**Comment by Rosalie241** [comment]
@philipjohnbarlow I'm sorry to uh, reply

but,

that video shows 0 proof that it's actually running on linux
and if it was real, it'd be pretty big news already.

so I'm sure it's fake and you shouldn't trust a video like that which doesn't show ANY evidence that it's actually what it claims to be.

**Comment by philipjohnbarlow** [comment]
Yup my thoughts as well, theres no pre-start up footage. Thought I'd share anyway just in case there was even remotely any validity in the claim.

**Comment by physios** [comment]
Doesn't battleeye work with ark?

**Comment by Lyle-Tafoya** [comment]
> Doesn't battleeye work with ark?

Ark has a native Linux build and uses the native Linux version of BattlEye. This thread is regarding compatibility with the Windows version of BattlEye via Proton.

**Comment by olealgoritme** [comment]
@kisak-valve Any updates on the BattlEye progression in Proton, or is Valve still ignoring it?

**Comment by kisak-valve** [comment]
Hello @olealgoritme, friendly reminder that I'm a moderator for Valve's issue trackers and not a Valve dev myself. I'm just as interested as everyone else to hear if/when there is progress with this, but we'll need to wait to hear from a Proton dev.

**Comment by olealgoritme** [comment]
Alright, let's wait some more.

**Comment by jsebean** [comment]
@philipjohnbarlow Check out looking glass here: https://github.com/gnif/LookingGlass
This is likely how the guy in the video achieved "linux support" of pubg
(tl;dr: It's not supported, it's a VM with direct passthrough access to the GPU so it will run, looking glass allows you to see what the passthrough GPU renders on the host OS)

**Comment by ghost** [comment]
Just commenting to say i'd love to see PlanetSide 2 support.

**Comment by Preisschild** [comment]
Since we got Arma 3 running well enough over Proton, BE is the only thing missing.

+1 for BE Proton support

Pls BE devs, make it happen.

**Comment by jarrard** [comment]
Yes ARMA3 seems to work quite nicely, but the BE driver doesn't initialize thus servers using it just kick you :(

I'm going to try wine4.6 with my custom proton compat profile once a lutris build comes out (I just copy over the wine files to my proton profile I made).

**Comment by sTiKyt** [comment]
BattleEye also will be useful to run "Heroes & Generals" using proton

**Comment by Faalagorn** [comment]
Survarium seems to work now (at least under proton-tkg), so it's just BattleEye that's preventing it running I guess.

**Comment by KAISTEP** [comment]
I hope this will give you hope:
https://www.reddit.com/r/linux_gaming/comments/bn2t8v/battleye_is_working_with_valve_to_add_support_for

https://imgur.com/a/g8CJQL5

**Comment by iklevente-zz** [comment]
There is some hope:
https://www.reddit.com/r/linux_gaming/comments/bn2t8v/battleye_is_working_with_valve_to_add_support_for

https://imgur.com/a/g8CJQL5

**Comment by SpookySkeletons** [comment]
So Hey, I have a question if any wine developer or Valve employee happens by...

I know wine used to work with battleye once some SMART and kernel access stuff was patched in until they adapted and started recognizing wine's presence, it allowed the anticheat to wrap the executable successfully as far as I could see and see you were on some real hardware.
With the modifications to the wine backend that build more DLLs as a Windows PE is this a full solution to masking wine's truck sized signature? Is this all that needs to be done to "pretend" you're windows or is there still a mountain ahead?

**Comment by Preisschild** [comment]
big update: With proton 4.11-1 battleye works in arma.

Thanks valve & be

**Comment by jarrard** [comment]
really? nice.   I dunno if it works in empyrion yet, gotta do some major testing this week as got a while new CPU/RAM cooling kit.

as it turns out, Empyrion uses EAC

**Comment by Preisschild** [comment]
Almost everyone seems to experience crashes exactly 15 minutes after joining a BattlEye enabled arma server.

**Comment by jarrard** [comment]
Maybe there is something that can be logged?   battleye log?

**Comment by yelhouti** [comment]
@Preisschild a PUBG enabled server? or are you talking abour arma? after installing pubg using steam I don't even get a loading screen. (Proton 4.11-2)

**Comment by jarrard** [comment]
Yeah I got that 15min crash with BE ARMA3 server.

**Comment by yelhouti** [comment]
XD, I was so excited thinking that I can no play PUBG on Linux..., any idea on when they will do the same to PUBG?

**Comment by JohnnyThunder2** [comment]
In regards to Arma 3, some people are able to stay in the server beyond the 15 minute mark. I've tried a bunch of different things to see if I can get it to work. Changing Kernals to match theirs, graphics settings, forwarding ports from my router and disabling firewalls. Steam beta on and off. I've also tried reinstalling battleye probably 5 times already. I installed some obscure "Fix" from a youtube channel for windows users having trouble. Nothing has worked for me so far, I'm gonna keep reinstalling battleye as sometimes windows users seem to have the problem of getting kicked every 15 minutes, and their fix is just to reinstall battleye until it works again.

**Comment by haterino** [comment]
Interessing enough i'm trying to work on a solution for battleye while playing escape from tarkov under lutris using proton 4.11. As far as it go, just locks at "Starting Battleye Service..." while it initializes, when the process BEservice.exe is killed, it goes that installed successfully and presents a new error "Failed to initialize BattlEye Service: Driver Load Error (31)."

**Comment by Vinjul1704** [comment]
@haterino nice to see some more escapers trying to get it running. Me and a few others got a discord server where we share some ideas and try to get it running (didn't do too much since the BE update has been released, but got quite far with getting rid of the startup lag and crashes before). I will send the link as soon as the owner says he is fine with that.

**Comment by Vinjul1704** [comment]
Here it is: https://discordapp.com/invite/uASUreY

**Comment by JanGalek** [comment]
Any news ? :)

**Comment by Preisschild** [comment]
@JanGalek it works for a few people on few servers in arma 3

**Comment by jarrard** [comment]
EAC and BE driver injection/loading is probably something that needs fixing at the WINE level, hopefully  valve devs can submit some fixes for it, we have already seen a few which is why it sometimes works atm.

**Comment by mkljczk** [comment]
Not just at WINE level, also might need some work at Linux kernel level… Actually, the best way would be to work with anti-cheats authors (they do), but implementing more and more low-level APIs is always a good thing

**Comment by redpiller** [comment]
There is a C api for such things, the only thing wine needs to to is to hook up to the kernel API and translate the calls as it would normally do with any other program

**Comment by wonkersbonkers** [comment]
would it be possible for developers to support the native battle eye on linux but the game is running in proton so they don't need to port the entire game? like no man's sky its not a linux game but the developers update it with proton in mind

**Comment by redpiller** [comment]
It wouldn't be the most streamlined aproach...

**Comment by jarrard** [comment]
They would need to put some code in that detects proton/wine/linux and thus plugs in the linux battleye libraries instead. Not sure how well that would work. Developers are unlikely to do this anyway. 

Perhaps a better approach for proton would be to detect battleye dll's and redirect to the local linux versions instead.

**Comment by ipimpat** [comment]
Please fix for Arma 3

**Comment by Saladin1812** [comment]
Please fix for Rainbow Six Siege

**Comment by HadiModarres** [comment]
ive found a fix, install windows

**Comment by redpiller** [comment]
Somebody block this troll

**Comment by bionade24** [comment]
> @JanGalek it works for a few people on few servers in arma 3

With the server I want it still crashes immeadeately.

**Comment by ZiTAL** [comment]
I have managed to start fortnite in debian:

launch lutris from console:

```
lutris 
```

Iniciate epic games and launch fortnite or other game that use this fucking shit, well...

If it doesn' work, kill it, let's list epic games program:

```
ps aux | grep Epic
zital     3763 10.0  0.0      0     0 ?        Zsl  23:07   0:13 [EpicGamesLaunch] <defunct>
zital     3763  4.6  4.5 4286528 319808 ?      Sl   23:07   0:05 C:/Program Files (x86)/Epic Games/Launcher/Engine/Binaries/Win64/UnrealCEFSubProcess.exe --type=renderer --no-sandbox --primordial-pipe-token=AA92F9BB6F4D59105D3262AE6941BC41 --lang=en-US --lang=en --locales-dir-path=C:/Program Files (x86)/Epic Games/Launcher/Engine/Binaries/ThirdParty/CEF3/Win64/Resources/locales --log-file=C:/users/zital/Local Settings/Application Data/EpicGamesLauncher/Saved/Logs/cef3.log --log-severity=warning --product-version=EpicGamesLauncher/10.15.8-13179977+++Portal+Release-Live UnrealEngine/4.23.0-13179977+++Portal+Release-Live Chrome/59.0.3071.15 --resources-dir-path=C:/Program Files (x86)/Epic Games/Launcher/Engine/Binaries/ThirdParty/CEF3/Win64/Resources --enable-pinch --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,3553;0,11,3553;0,12,3553;0,13,3553;0,14,3553;0,15,3553;0,16,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,3553;1,11,3553;1,12,3553;1,13,3553;1,14,3553;1,15,3553;1,16,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,3553;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,3553;2,11,3553;2,12,3553;2,13,3553;2,14,3553;2,15,3553;2,16,3553;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,3553;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,3553;3,13,3553;3,14,3553;3,15,3553;3,16,3553;4,0,3553;4,1,3553;4,2,3553;4,3,3553;4,4,3553;4,5,3553;4,6,3553;4,7,3553;4,8,3553;4,9,3553;4,10,3553;4,11,3553;4,12,3553;4,13,3553;4,14,3553;4,15,3553;4,16,3553 --disable-accelerated-video-decode --disable-webrtc-hw-vp8-encoding --disable-gpu-compositing --service-request-channel-token=AA92F9BB6F4D59105D3262AE6941BC41 --renderer-client-id=3 --mojo-platform-channel-handle=1732 /prefetch:1
zital     4070  0.0  0.0   6080   892 pts/1    S+   23:09   0:00 grep Epic

```

well I need to kill the 3763 processes:

```
kill -9 3763 3763
```

kill lutris too:

```
ps aux | grep lutris
zital     3657  2.2  1.7 695900 120672 pts/0   Sl+  23:06   0:01 /usr/bin/python3 /usr/games/lutris
zital     3657  2.3  0.2  35204 19544 pts/0    S    23:07   0:01 lutris-wrapper: Epic Games Store
zital     3657  3.4  0.2  18580 17564 ?        Ss   23:07   0:03 /home/zital/.local/share/lutris/runners/wine/lutris-5.6-2-x86_64/bin/wineserver

kill -9 3657 3657 3657
```

try again

**Comment by sharkguto** [comment]
any update??? It would be great if we can play PUBG on proton

**Comment by jarrard** [comment]
I don't think so. We are all waiting for some form of support to EAC and Battleye.  

There is a experimental Kernel patch for Wine stuff that may allow for anti-cheat software to work in the future if merged. 
I don't think its been accepted or merged into mainline yet, once that happens it will be news worthy.

**Comment by TheAifam5** [comment]
@jarrard link please?

**Comment by Titaniumtown** [comment]
lets hope it gets merged! I haven't played pubg since it launched lol. It's just sitting in my library

**Comment by wonkersbonkers** [comment]
is there a repository for this experimental kernel

**Comment by Titaniumtown** [comment]
I would like to build it

**Comment by kisak-valve** [comment]
Hello @Titaniumtown, @wonkersbonkers, I think there's been some misinformation that came out of the community discussion surrounding this [patch](https://lore.kernel.org/lkml/20200530055953.817666-1-krisman@collabora.com/). Mainly, there was some discussion about an alternative approach to solving the technical riddle and that is known to be incompatible with anti-cheat software and https://lore.kernel.org/lkml/9a512096-7707-3fc6-34ba-22f969c0f964@gmail.com/ got a distorted interpretation by some onlookers.

This patch is exploring a potential performance optimization and the developers involved don't appear to be interested in using an alternative solution which would be going backwards with regard to eventually getting anti-cheat mechanisms working. It's not directly related to supporting anti-cheat software.

(Fair disclaimer this is also a second-hand interpretation and should not be used as a primary reference.)

If this isn't the patch in question, feel free to clarify.

**Comment by StidOfficial** [comment]
BattlEye work for 1 month and last 2 days ago, i have error "Failed to initialize BattlEye Service: Driver Load Error (31)." with Arma 3 on Proton 5.0-9.

**Comment by Titaniumtown** [comment]
That's very interesting @StidOfficial hmm

**Comment by IvanCharcos** [comment]
Just tried it with PubG. Same error.

**Comment by batara666** [comment]
BlackSquad please

**Comment by bionade24** [comment]
> BlackSquad please

What do you mean?

**Comment by sharkguto** [comment]
> > BlackSquad please
> 
> What do you mean?

i believe it is a game from steam that's use battleeye as anti cheat rsrs :laughing:

**Comment by StidOfficial** [comment]
Probably the best way to fix it : `<Link removed by moderator>`

**Comment by kisak-valve** [comment]
Hello @StidOfficial, no, that workaround is plain cheating. There's nothing there that could be considered a fix which can be used to improve support in Proton.

**Comment by bionade24** [comment]
> Hello @StidOfficial, no, that workaround is plain cheating. There's nothing there that could be considered a fix which can be used to improve support in Proton.

He not even said it's a workaround or even a solution, just the best way in the sad reality.

**Comment by mkljczk** [comment]
Not to improve support in Proton and I don't encourage anyone to cheat, but would be curious to see whether the battleye-protected games would work without other issues after bypassing this piece of ~~shi~~ of battleye

**Comment by bionade24** [comment]
> Probably the best way to fix it : `<Link removed by moderator>`

Really sad this Link gets removed. Sad that you're supporting a system repressing you.

**Comment by Eeems** [comment]
> > Probably the best way to fix it : `<Link removed by moderator>`
> 
> Really sad this Link gets removed. Sad that you're supporting a system repressing you.

You do realize this is an official Valve repository right?

**Comment by mkljczk** [comment]
> > Probably the best way to fix it : `<Link removed by moderator>`
> 
> Really sad this Link gets removed. Sad that you're supporting a system repressing you.

@bionade24 they work for Valve. They would rather get a solution that would both satisfy BattlEye by giving them root-level access to our Linux machines and provide us with working games

**Comment by bionade24** [comment]
At WineHQ this hopefully gets discussed more open: https://bugs.winehq.org/show_bug.cgi?id=37355
Edit: Maybe more appropriate than spamming into system internals: https://bugs.winehq.org/show_bug.cgi?id=45124

**Comment by mkljczk** [comment]
> At WineHQ this hopefully gets discussed more open: https://bugs.winehq.org/show_bug.cgi?id=37355

Not sure. Shortly after Guy1524 published his set of patches that correctly implement all the checks required by EAC, they made some server-side changes to prevent EAC from working with Wine. BattlEye would likely do the same

**Comment by bionade24** [comment]
> > At WineHQ this hopefully gets discussed more open: https://bugs.winehq.org/show_bug.cgi?id=37355
> 
> Not sure. Shortly after Guy1524 published his set of patches that correctly implement all the checks required by EAC, they made some server-side changes to prevent EAC from working with Wine. BattlEye would likely do the same

:(
Don't know, but the hack @StidOfficial provided attacked them at IPC level. Probably harder to patch permanently.

**Comment by bionade24** [comment]
A realistic solution to this problem: Linux gets a  working FLOSS anti-cheat for Linux and the Games only work if the binary is signed. A realistic compromise?

**Comment by batara666** [comment]
oh cmon, we all know, Linux user are not cheater

**Comment by Titaniumtown** [comment]
lol

**Comment by bionade24** [comment]
> oh cmon, we all know, Linux user are not cheater

Yeah, even the devs probably know that Linux users just wanna enjoy their spare time but we're live in a world where management doesn't care. Or maybe it's because they know that we know that calculating all physics on the server would make cheating impossible but won't do that for profit reasons.

**Comment by darutoko** [comment]
> > oh cmon, we all know, Linux user are not cheater
> 
> Yeah, even the devs probably know that Linux users just wanna enjoy their spare time but we're live in a world where management doesn't care. Or maybe it's because they know that we know that calculating all physics on the server would make cheating impossible but won't do that for profit reasons.

I am very curious, can you please explain how calculating all physics on the server would make impossible aim-bots, wall hacks, map hacks etc.?

**Comment by Saladin1812** [comment]
@darutoko  it's harder for the player to make changes on the server side than the client side.. a player can easily change stuff on the client side.. however a server side anti cheat can't detect changes made on the client side but can detect stuff going on in the server because of the client side changes/hacks.. but for that server side anti cheats should be very strong also using proxy is very great on server side.. so that if a player tried to figure out the server address.. he would only see the proxy server address not the actual server address.. however server side anti cheat has its own risks of someone figures out to make client side changes that server side won't be able to detect it can caus alot of issues.. it's like having high risk high reward.

**Comment by darutoko** [comment]
> @darutoko it's harder for the player to make changes on the server side than the client side.. a player can easily change stuff on the client side.. however a server side anti cheat can't detect changes made on the client side but can detect stuff going on in the server because of the client side changes/hacks.. but for that server side anti cheats should be very strong also using proxy is very great on server side.. so that if a player tried to figure out the server address.. he would only see the proxy server address not the actual server address.. however server side anti cheat has its own risks of someone figures out to make client side changes that server side won't be able to detect it can caus alot of issues.. it's like having high risk high reward.

I dont' understand why you wrote this. In this text wall there is no explanation of how calculating all physics on the server would make impossible aim-bots, wall hacks, map hacks etc. How for example wall hack (texture and model manipulation) is affected by physics calculations?

**Comment by bionade24** [comment]
> > > oh cmon, we all know, Linux user are not cheater
> > 
> > 
> > Yeah, even the devs probably know that Linux users just wanna enjoy their spare time but we're live in a world where management doesn't care. Or maybe it's because they know that we know that calculating all physics on the server would make cheating impossible but won't do that for profit reasons.
> 
> I am very curious, can you please explain how calculating all physics on the server would make impossible aim-bots, wall hacks, map hacks etc.?

The server can determine which actions can physically be done and which not. (E.g when you went trough a wall you get banned.) Or when you aiming is way to good for a human (human can have good aim but will have at least certain stuttering on the target. Finally, you can calculate which opponent a player could have seen when. Tactical cheating would still be possible, but you can also e. g. do that in CSGO when you're in Teamspeak with a dead player giving tips.

**Comment by darutoko** [comment]
> > > > oh cmon, we all know, Linux user are not cheater
> > > 
> > > 
> > > Yeah, even the devs probably know that Linux users just wanna enjoy their spare time but we're live in a world where management doesn't care. Or maybe it's because they know that we know that calculating all physics on the server would make cheating impossible but won't do that for profit reasons.
> > 
> > 
> > I am very curious, can you please explain how calculating all physics on the server would make impossible aim-bots, wall hacks, map hacks etc.?
> 
> The server can determine which actions can physically be done and which not. (E.g when you went trough a wall you get banned.) Or when you aiming is way to good for a human (human can have good aim but will have at least certain stuttering on the target. Finally, you can calculate which opponent a player could have seen when. Tactical cheating would still be possible, but you can also e. g. do that in CSGO when you're in Teamspeak with a dead player giving tips.

@bionade24 lol
Your answer is even funnier than i thought it would be, thank you.

**Comment by StidOfficial** [comment]
I shared this link for create a pipeline to the actual battleye linux client (because he doesn't need kmod like bedaisy.sys).
This link doesn't talk about cheating but emulation of BattlEye and this informations can help us to port BattlEye on proton (wine).

If the issue is locate on "MmMapLockedPagesSpecifyCache" good new.

**Comment by Beast42** [comment]
Problem still not resolved?

**Comment by kiwy42** [comment]
I so much whises this issue will find a proper solution in the future.
Steam and BattleEye seemed to have been talking to each other but so far... no nothing.
The best thing would be to not allow any kernel level anti-cheat for steam game thus avoiding any to strong application installed and make the proton compatibility so much simple. 
I do not like cheaters in any online game as much as the next fellow but we have to be true about it. No anticheat has been anything but effective... being CoD Warzone, PUBG or Fortnite no one succeed against cheaters despite very intrusive detection software... game developers should focus on server anticheat instead of trying to install spyware on our hardware. 
They will never be able to control user hardware even on console the cheat is placed between the gamepad and the console...

**Comment by Storm-Engineer** [comment]
In ARK: Survival Evolved, this is an even bigger problem because the game's Linux build is completely broken and unusable, and the developer has absolutely no intentions to fix it. BattleEye works in said  Linux build, except the game itself is unplayable in many ways... so Linux users are forced to run the game via Wine instead.

**Comment by r4jk3** [comment]
I have created a steam support ticket and got redirected to this page. @Storm-Engineer said all what i had to say actually. We need fix for this BatteEye thing i sense its the only thing blocking Linux community to prosper in gaming industry. Proton is a mayor thing that happened to us and because of it i finally got rid of Windows ( i will not return to it even at this stage ). I have yesterday sent a email to BattleEye support team about it but unfortunately still no response.

**Comment by linuxergr** [comment]
BattleEye should be installed by default, when games are installed on Linux, ever since it concerns paid games (not free of charge) like Insurgency and Day of Infamy

**Comment by RevoluPowered** [comment]
Perhaps valve should look into banning new game releases which contain battle eye and release a formal statement about it. This will kill Battle Eye's new market opportunities.

Treat them the way they treat other platforms which are blocked by their software.

It's clear they're not invested in a developer solution or even offering a branch which could be pursued by bigger companies if they need Linux support even if they offer financial incentives.

**Comment by mkljczk** [comment]
@RevoluPowered that would only make other stores more competitive

**Comment by Spartan322** [comment]
> Perhaps valve should look into banning new game releases which contain battle eye and release a formal statement about it. This will kill Battle Eye's new market opportunities.
> 
> Treat them the way they treat other platforms which are blocked by their software.
> 
> It's clear they're not invested in a developer solution or even offering a branch which could be pursued by bigger companies if they need Linux support even if they offer financial incentives.
@RevoluPowered 

That's not only a horrible business strategy, but that harms Valve's economic model. Its also horribly anti-consumer, direct opposition to Linux methodology, (being foremost volunteerism) and a vengeful solution would make them just as ethically and morally repugnant as every single big name publisher that everyone hates. Going down that road is a slippery slope especially in business and after you open that can of worms once, you can almost never close it.

**Comment by RevoluPowered** [comment]
> That's not only a horrible business strategy, but that harms Valve's economic model. Its also horribly anti-consumer, direct opposition to Linux methodology, (being foremost volunteerism) and a vengeful solution would make them just as ethically and morally repugnant as every single big name publisher that everyone hates. Going down that road is a slippery slope especially in business and after you open that can of worms once, you can almost never close it.

Good point I agree, but for now what are the alternatives?

**Comment by RevoluPowered** [comment]
I toyed with making an executable tool to strip / jump across battle eye entirely so people could actually run the game on linux.

I think for games like Arma 3 you could do this without problems because battle eye can be disabled as a requirement on servers, but some other games which rely on their servers it wouldn't work / fortnite.

**Comment by Storm-Engineer** [comment]
Is anyone actually working on this?

**Comment by xznhj8129** [comment]
THREE GOD DAMN YEARS

**Comment by Spartan322** [comment]
> Is anyone actually working on this?

Really no reason to ask this, this issue is clearly not dead, the problem has mostly been a Linux kernel problem before anything else, which is starting to make it more feasible to work on this with the latest experimental kernel version, but that just got released about a week ago, and running experimental kernel versions is not practical, most especially to dev on or play a game with, we have to wait until 5.11 is stable at least, shouldn't be long but its still going to take a bit, you gotta be patient man. (even then we probably won't see a change in Proton or Wine for a little while on this front until mainstream distros start defaulting to 5.11, or at least a while after its stable, kinda pointless to dev a feature heavily when you have nobody to test it and use it, especially when its bleeding edge)

**Comment by mirh** [comment]
Syscall User Dispatch has nothing to do with anticheat. 
And it's not a kernel problem before anything else. That's also a biggie to be sure, but first of all wine's ntoskrnl has to gain support for a lot of functions. 
And then maybe you can start poking at BE.

**Comment by TheAifam5** [comment]
Valve can't do anything much about it even if they somehow manage, there may be a great risk to be flagged/banned. As stated (i think), Valve will/work with BE but is up to BE to support such case. I feel that because it's opensource, and you can modify WINE as you like, that is a big concern for the BE.

**Comment by Spartan322** [comment]
> Syscall User Dispatch has nothing to do with anticheat.
> And it's not a kernel problem before anything else. That's also a biggie to be sure, but first of all wine's ntoskrnl has to gain support for a lot of functions.
> And then maybe you can start poking at BE.

While its not a specific target, its likely part of a puzzle piece solution, some anticheats operate on a similar level as DRM in this specific instance, (and I'm pretty sure BattleEye was one of the ones where this applied, though correct me if I'm wrong) and they've been hampered just as badly, (not to mention most mainstream invasive anticheat games also incorporate some kind of DRM anyway, or the anticheat is basically the DRM) so I think its worth mentioning.

**Comment by Storm-Engineer** [comment]
> > Is anyone actually working on this?
> 
> Really no reason to ask this, this issue is clearly not dead

I respectfully disagree. The issue is pretty old, has no assignee, and the last response from a Valve dev was 2019 January. As such, I have no way of knowing if this is being actively worked on, or it is being set aside because other issues are deemed a higher priority. And thus, I asked, hoping that if it is being worked on, a dev may reply to let us know.

The comments here are definitely active but that does not tell anything about if it is being worked on by the developers or not.

**Comment by Storm-Engineer** [comment]
> Valve can't do anything much about it even if they somehow manage, there may be a great risk to be flagged/banned. As stated (i think), Valve will/work with BE but is up to BE to support such case. I feel that because it's opensource, and you can modify WINE as you like, that is a big concern for the BE.

Other companies and anti-cheat systems have absolutely no problem with Wine players, and this includes both Valve Anti-Cheat and Blizzard Anti-Cheat. The whole notion that it would somehow be easier to cheat with Wine, is just not true - unless we are talking about some extremely bad and shitty anti-cheat. If it is well written - and given BE is widely used I assume that for the most part it is, or at least adequately written - then Wine should pose no threat.

And I mean, BE is not trying to block Wine users from playing, they just doN't care to put effort into supporting it. (The only anti-cheat I personally know of that purposefully blocks Wine users is Epic Anti-Cheat... but then Epic is an asshole company as we all know.)

And Wine is not trying to trick anyone, while it mimics the Windows environment for programs to work, it does not try to pretend to be real Windows. Programs can easily see that they are being run under Wine, and behave accordingly.

**Comment by mirh** [comment]
EAC doesn't block wine. They just have a check to load their special gimped wine-specific version that some games have opted into (which of course fails when instead the developer didn't). 
And just like BE they do support native linux anyway. 

It's just that the authentication/integrity infrastructure windows has is nowhere to be seen. Wine by itself runs entirely in userspace, making bypass super easy, and you can't blame anyone here. 
It's not impossible to solve this, but it's super complex and not even worth discussing when anyway wine is so much broken in supporting the windows kernel apis. 
So please, just stop spamming. And wait.

**Comment by jarrard** [comment]
SQUAD and MORTAL2ONLINE with EAC work with Proton, but you may need to download the verification files yourself as there is a bug in proton where EAC tool fails to fetch the verification files, and can even delete them for some reason.

As for BE, yep its a BIG BIG issue, I want to play games like Tarkov and ARMA3 but can't because I cannot figure out how BE works.

EAC FYI has OS flags on their server which devs can enable. Linux is WINE64, Windows is WIN64, and MacOS is probably OSX or something... I guess there is ARM ones and others also. The EAC files of WINE64 and WIN64 are identical btw, you can grab them from your windows EAC folders and copy them right over. 


PS. _None of these Anti Cheat methods seem to fully stop hackers/cheaters, I've talk to these people on Discord and its still very easy to work around these anticheat methods as many have probably suspected. Sure it can require the simpletons from cheating, but as hack/cheat tools (all on windows btw, Linux is not source of hackers/cheaters) advance they also make it easier for people to do._

**Comment by RealJTG** [comment]
Vote with your money, don't buy games that requires BE.
Kernel level anti-cheats/DRM _shouldn't be used_ on any OS (they don't work anyway, I'm talking about you Valorant).

**Comment by jarrard** [comment]
> Vote with your money, don't buy games that requires BE.
> Kernel level anti-cheats/DRM _shouldn't be used_ on any OS (they don't work anyway, I'm talking about you Valorant).

This statement is not helpful. A huge number of people already own these games long before moving to Linux. Also EAC has taken steps to help devs support wine/proton/linux (optional) and there is a Linux based BE client with hopefully more support down the line for proton.

**Comment by Lamdarer** [comment]
https://www.protondb.com/app/218230 https://github.com/ValveSoftware/Proton/issues/5295
Even though no official statement of the developers, the game currently runs fine using the latest Proton-GE version with Proton BattlEye Runtime installed.

**Comment by jarrard** [comment]
Yes I tested PS2 out it works. However native Proton Exp builds have a issue with it's launcher and will display a white or black box. (you can still blind click areas to activate stuff)

When I tested it I used protonGE6.8 to install as it solved the launcher issue, then I switched to proton-exp and launched the game (blind click) and it worked fine. 
Hopefully launcher fixes start getting applied to mainline proton more such as this one and the BSG launcher of Tarkov (it requires dotnet48 and wined3d to see atm unfortunately).

I don't think dotnet is needed for EFT, just some mono patching and dxvk fixes to render launcher correctly.

**Comment by Atemu** [comment]
@jarrard would be better to put discussion about PS2-specific issues in the dedicated issue #5295.

**Comment by csedesio** [comment]
hey guys i am trying to run the epic's version of ark on wine 7.0 but it says this Failed to initialize BattlEye Service: Driver Load Error (1053). As far as i know ark should work with battleye

**Comment by HolimaX** [comment]
... still not fixed .. holy crap.

BTW, Destiny 2 via Steam on FC37 also affected.

**Comment by Atemu** [comment]
This issue is as much fixed as it can be. It's up to the individual game devs to enable Linux support. Destiny's developer explicitly decided against supporting Linux/the Steam Deck. If you want to complain, direct your anger towards them.

Planetside 2's developer did enable it for example and BattleEye works flawlessly. Better than on windows actually.

**Comment by Faalagorn** [comment]
There's still some games, like Escape from Tarkov, that lacks the required function to get everything working (in case of Tarkov it's playing online), but yeah.

**Comment by landjalan** [comment]
for me this issue is pretty easy. just don´t buy/play a game which does support linux one way or another :) there is literally 10 thousands of games out there that all run on linux... if a specific title of a specific genre does not work, just take a competitor... EASY

**Comment by nipnipj** [comment]
Does battleeye runtime allows you to play PUBG these days?

**Comment by Atemu** [comment]
PUBG doesn't allow Battleye and its two other anticheats to run on Linux.

**Comment by HolimaX** [comment]
> for me this issue is pretty easy. just don´t buy/play a game which does support linux one way or another :) there is literally 10 thousands of games out there that all run on linux... if a specific title of a specific genre does not work, just take a competitor... EASY

Yes, all or nothing approach is legit, but ... still .. it is in humans nature to desire for things :)
